feat(#10221): add documentation to create and update person, place and report (#1962)
medic/cht-docs: content/en/building/reference/api.md

}
```

### POST /api/v1/person

> 

Reviewer: issue: add a `"added in X.X.X"` version note for all the endpoints

> 

Reviewer: issue: add the description of the endpoint

#### Supported Properties

Use JSON in the request body to specify a person’s details.

#### Permissions

> Either of the two:
> 1. `can_edit` 
> 2. `can_view_contacts` and `can_create_people`

Reviewer: ```suggestion

Should have `can_edit` or both `can_view_contacts` and `can_create_people`
```

The wording seemed a bit confusing to me.

#### Required
| Key  | Description                         |
| ---- | ----------------------------------- |

> | name | String used to describe the person. |

Reviewer: ```suggestion
| name | Name of the person. |
```

| type | ID of the `contact_type` for the new person. Pass in `person` for older versions. |
| parent | ID of the parent document, which is a place, for the new person. The type of the parent must belong to one of the allowed `parents` for the `contact_type` id in the settings config. |

#### Optional

| Key           | Description                                                            |
| ------------- | ---------------------------------------------------------------------- |
| reported_date | Timestamp of when the record was reported or created. Defaults to now. |

> | _id | ID of the new person document to be created. |

Reviewer: ```suggestion
| _id | ID of the new person document to be created in UUID format. |
```

Reviewer: suggestion: how about adding another column that contains the type of the field instead of doing a "... string format" in the description?

Author: Hmm makes sense.

| short_name | String to denote short name of the person. |
| phone | Phone number of the person in string format. |

> | role | Role of the person in string format. |

Reviewer: suggestion: let's use `fields` instead of `key` here

#### Examples
Create a person with a clinic as its parent place.

```bash
POST /api/v1/person
Content-type: application/json

{
    "name": "dummyuser",
    "type": "person",
    "parent":"01992f01-f155-4386-8538-5080c3155585"
}
```

Example response:
```json
{
    "_id": "4dcd842e813fd1bcabec03f98f002a98",
    "_rev": "1-bf4ecb779a3a369f6162a149853cbb02",
    "name": "dummyuser",
    "type": "contact",
    "parent": {
        "_id": "01992f01-f155-4386-8538-5080c3155585",
        "parent": {
            "_id": "35a2f31e-705c-4833-b385-efd069b1ce3f",
            "parent": {
                "_id": "29368c93-d267-4e80-8fbe-6543f702ff30"
            }
        }
    },
    "reported_date": 1755519112122,
    "contact_type": "person"
}
```

> ### PUT /api/v1/person

Reviewer: ```suggestion

### PUT /api/v1/person
```

#### Supported Properties

Use JSON in the request body to specify a person’s details.

#### Permissions
Either of the two:
1. `can_edit` 
2. `can_view_contacts` and `can_update_people`

#### Required immutable fields
| Key  | Description                         |
| ---- | ----------------------------------- |
| _id | ID of the person doc to be updated. |
| _rev | Revision ID of the person doc to be updated. |
| reported_date | Timestamp of when the record was reported or created. |
| contact_type| Required if the type of the person is `contact`. |
| type | The type of the person. |

> | parent | The parent lineage of the person to be updated, can be hydrated or dehydrated. |

Reviewer: issue: `parent` cannot be updated

Reviewer: I think it would be helpful to document why `parent` cannot be updated

Author: I meant "person to be updated"'s parent

}
```

### POST /api/v1/person

> *Added in x.x.x*

Reviewer: For now you can add the version as `5.0.0`. If something changes, it can be changed.

        }
    },
    "contact": {
        "_id": "00058058-e637-4e65-b5db-cd4b2da4375e",
        "parent": {
            "_id": "0fc7c7c4-76c0-4c1b-8eac-43ee5b42f646",
            "parent": {
                "_id": "c79e489a-1ee0-4537-a618-bfab496a7e29",
                "parent": {
                    "_id": "a177578c-a2a2-4731-931c-2bbff842d0f4"
                }
            }
        }
    },
    "reported_date": "2025-08-19T14:48:16.436Z",
    "contact_type": "clinic"
}
```

Response:
```json
{
    "_id": "36aed043a50315f00f625af18c004bcf",
    "_rev": "2-7d9114d21f80451b436d6d25d8ac5a75",
    "name": "new place",
    "type": "contact",
    "parent": {
        "_id": "35a2f31e-705c-4833-b385-efd069b1ce3f",
        "parent": {
            "_id": "29368c93-d267-4e80-8fbe-6543f702ff30"
        }
    },
    "contact": {
        "_id": "00058058-e637-4e65-b5db-cd4b2da4375e",
        "parent": {
            "_id": "0fc7c7c4-76c0-4c1b-8eac-43ee5b42f646",
            "parent": {
                "_id": "c79e489a-1ee0-4537-a618-bfab496a7e29",
                "parent": {
                    "_id": "a177578c-a2a2-4731-931c-2bbff842d0f4"
                }
            }
        }
    },

>     "reported_date": "2025-08-19T14:48:16.436Z",

Reviewer: ```suggestion
    "reported_date": 1756277807624,
```

}
```

### POST /api/v1/report

*Added in 5.0.0*
#### Description
Used to create a report.

#### Supported Properties

Use JSON in the request body to specify a report’s details.

#### Permissions
Should have `can_view_reports` and `can_create_records`.

#### Required
| Field   | Description                                                                                                      | Format     |
|---------|------------------------------------------------------------------------------------------------------------------|------------|
| form    | Must be a valid form value from the forms available in the `medic-client/doc_by_type` view (queried with `key=["form"]`). | string     |
| type    | Type of the report.                                                                                              | string     |
| contact | ID of the contact document (can be either a person or a place) for the new report.                               | UUID string |

> Note: A valid `form` value, `type` being `data_record` and having a valid `contact` is necessary for the report to appear in the webapp.

Reviewer: ```suggestion
*Note: A valid `form` value, `type` being `data_record` and having a valid `contact` is necessary for the report to appear in the webapp.*
```

| reported_date | Timestamp of when the record was reported or created. Defaults to `now`.    | `YYYY-MM-DDTHH:mm:ssZ`, `YYYY-MM-DDTHH:mm:ss.SSSZ`, or unix epoch |
| _id           | ID of the new report document to be created.                                | UUID string                                      |
| fields        | Fields containing the report data.                                          | Object                                           |

#### Examples
Request:
```bash
POST /api/v1/report
Content-type: application/json

{
    "form":"pregnancy_home_visit",
    "type":"data_record",
    "contact":"00ba941a-a052-4e45-ada7-c32f59542133"
}
```

Response:
```json
{
    "_id": "b8208fa332bf1f09b606e6efd8002a4a",
    "_rev": "1-9ffca0e670bcc111de86f68ae8f47d3b",
    "form": "pregnancy_home_visit",
    "type": "data_record",
    "contact": {
        "_id": "00ba941a-a052-4e45-ada7-c32f59542133",
        "parent": {
            "_id": "4583e5aa-b1c1-40d4-9f65-ce63610b8c52",
            "parent": {
                "_id": "f0d44e4c-bd37-44a2-a7a5-dd38de32d9a2"
            }
        }
    },

>     "reported_date": "2025-08-24T11:37:06.815Z"

Reviewer: ```suggestion
    "reported_date": 1755490087000
```

}
```

> ### PUT /api/v1/report
> 
> #### Description

Reviewer: ```suggestion
### PUT /api/v1/report

*Added in 5.0.0*

#### Description
```

            "parent": {
                "_id": "f0d44e4c-bd37-44a2-a7a5-dd38de32d9a2"
            }
        }
    },
    "reported_date": "2025-08-24T11:37:06.815Z"
}
```

Request:
```bash
PUT /api/v1/report
Content-type: application/json
{
    "_id": "b8208fa332bf1f09b606e6efd8002a4a",
    "_rev": "1-9ffca0e670bcc111de86f68ae8f47d3b",
    "form": "pregnancy_danger_sign",
    "type": "data_record",
    "contact": {
        "_id": "00ba941a-a052-4e45-ada7-c32f59542133",
        "parent": {
            "_id": "4583e5aa-b1c1-40d4-9f65-ce63610b8c52",
            "parent": {
                "_id": "f0d44e4c-bd37-44a2-a7a5-dd38de32d9a2"
            }
        }
    },
    "reported_date": "2025-08-24T11:37:06.815Z"
}

```

Response:
```json
{
    "_id": "b8208fa332bf1f09b606e6efd8002a4a",
    "_rev": "2-82b274ab67cf5fbab0ea870cbb62ab48",
    "form": "pregnancy_danger_sign",
    "type": "data_record",
    "contact": {
        "_id": "00ba941a-a052-4e45-ada7-c32f59542133",
        "parent": {
            "_id": "4583e5aa-b1c1-40d4-9f65-ce63610b8c52",
            "parent": {
                "_id": "f0d44e4c-bd37-44a2-a7a5-dd38de32d9a2"
            }
        }
    },

>     "reported_date": "2025-08-24T11:37:06.815Z"

Reviewer: ```suggestion
    "reported_date": 1755490087000
```
